### PR TITLE
[ADD] extended account selectors

### LINKED ISSUE #4: extended account selectors
_From @sbidoul on September 3, 2017 13:9_

Accounting expressions in mis_builder have the form

`(bal|crd|deb)(p|i|e)[account selector]([optional move line domain])?`

The account selector is currently based on account codes (eg `70%`).

Further account filtering can be added in the move line domain, eg `('account_id.type', '=', ...)`. But that is slow because additional database queries must be done, whereas the account selector boils down to one `read_group` per period and move line domain.

So it is desirable to avoid using the move line domains to filter on accounts.

To implement this, the account selector can be extended to support any domain on gl accounts.

_Copied from original issue: OCA/account-financial-reporting#345_

### PR DESCRIPTION
**Pull request by sbidoul**
closes #4

### CODE REVIEW

**Comment by sbidoul** [comment]
@jbeficent would you like to test this one?

File: mis_builder/models/aep.py [hunk 1]
```diff
@@ -84,101 +90,105 @@ def __init__(self, companies, currency=None):
...
+            # legacy bal_NNN%
+            acc_domain = self._account_codes_to_domain(account_sel[1:])
+        else:
+            assert account_sel[0] == '[' and account_sel[-1] == ']'
+            inner_account_sel = account_sel[1:-1].strip()
+            if not inner_account_sel:
+                # empty selector: select all accounts
+                acc_domain = tuple()
+            elif inner_account_sel.startswith('('):
+                # account selector is a domain
```
Comment (richard-willdooit) [inline mis_builder/models/aep.py]: The domain parsing is assuming the first element of the domain is a tuple - it could be a '!', '|', or '&'

**Review by richard-willdooit — CHANGES_REQUESTED** [review]
Only really the one suggested enhancement at the account domain...

File: mis_builder/models/aep.py (hunk 1 above)
Comment (sbidoul) [reply mis_builder/models/aep.py]: Right! Thanks for the in depth review!

Comment (sbidoul) [reply mis_builder/models/aep.py]: Fixed at 0314b8451a0b12228699f0ed4f5b7f329bcec5d4

**Review by richard-willdooit — APPROVED** [review]
Cool.

**Comment by sbidoul** [comment]
Small performance improvement to last commit and improved test.

**Review by adrienpeiffer — APPROVED** [review]
:+1:

**Pull request merged**